### PR TITLE
Check for all set-length-modifying methods in modified-iterating-set (close #9334)

### LINKED ISSUE #9334: Add checks for all set-length-modifying methods in modified-iterating-set
### Current problem

`modified-iterating-set` currently only checks for the `.add` and `.remove` methods, but not several other length-modifying methods.

### Desired solution

Add checks for the `.clear`, `.discard`, and `.pop` methods being called on a set which is being iterated over.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mdbernard**
- [x] Document your change, if it is a non-trivial one.
    * I would consider this change trivial, but this is my first PR to pylint, so please let me know if it would not be considered trivial by this repo's standards.
- [x] Relate your change to an issue in the tracker if such an issue exists
    * I did not find a related issue.
- [x] Write comprehensive commit messages and/or a good description of what the PR does.
- [x] Keep the change small, separate the consensual changes from the opinionated one.
  Don't hesitate to open multiple PRs if the change requires it. If your review is so
  big it requires to actually plan and allocate time to review, it's more likely
  that it's going to go stale.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Currently, `modified-iterating-set` only checks for when the `.add` or `.remove` methods are called on the set being iterated over. However, `.clear`, `.discard`, and `.pop` also raise `RuntimeError` when used on a set while it is being iterated over, so there is value in linting them as well.

Closes #9334

### CODE REVIEW

File: tests/functional/m/modified_iterating.py [hunk 1]
```diff
@@ -39,6 +39,22 @@
 for item in item_set:
     item_set.add(item + 10)  # [modified-iterating-set]
 
+item_set = {1, 2, 3}
+for item in item_set:
+    item_set.clear()  # [modified-iterating-set]
+
+item_set = {1, 2, 3}
+for item in item_set:
+    item_set.discard()  # [modified-iterating-set]
```
Comment (Pierre-Sassoulas) [inline tests/functional/m/modified_iterating.py:48]:
```suggestion
    item_set.discard(1)  # [modified-iterating-set]
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Great contribution, thank you !

**Comment by Pierre-Sassoulas** [comment]
Here's an example of changelog : https://github.com/pylint-dev/pylint/blob/main/doc/whatsnew/fragments/8701.feature (you can refer to this merge request as there is no issue associated afaik)

**Comment by mdbernard** [comment]
> Here's an example of changelog : https://github.com/pylint-dev/pylint/blob/main/doc/whatsnew/fragments/8701.feature (you can refer to this merge request with `Refs #9333` as there is no issue associated afaik)

(I made a new issue for this for ease of reference)

File: doc/whatsnew/fragments/9334.feature [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Check for `.clear`, `.discard`, and `.pop` methods being called on a set while it is being iterated over.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/9334.feature:1]:
```suggestion
Check for `.clear`, `.discard`, `.pop` and `remove` methods being called on a set while it is being iterated over.
```

**Review by Pierre-Sassoulas — DISMISSED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Thank you for your work, very nice first contribution :thumbsup: this is going to be released in 3.1.0, congratulations on becoming a pylint contributor ;)

**Pull request merged**